src/routes/comments.js

```diff
@@ -84,34 +109,54 @@ Create mongoose ObjectId from parameter
 Find and delete comment in db using findByIdAndDelete()
 Return confirmation message on success
 */
-router.delete("/:commentID", async (req, res) => {
-  Comment.findByIdAndDelete(new mongoose.Types.ObjectId(req.params.commentID))
-    .then(deleted => {
-      if (deleted) {
-        res.json({ deleted: deleted });
-      } else {
-        return Promise.reject("Error: Comment not found");
+router.delete('/:commentID', jwtAuth, (req, res) => {
+  Comment.findById(req.params.commentID)
+    .then(comment => {
+      if (!comment) {
+        return res.status(404).json({ Error: 'Comment not found' });
+      }
+
+      if (req.user._id !== comment.user.toString()) {
+        return res.status(403).json({ message: "This isn't your comment" });
       }
+
+      comment.remove().then(comment => res.status(204).end());
```

Comment:
Any reason to use `findById()` and `remove()` over `findByIdAndDelete()`? Just curious

Reply:
@a-w-m `findByIdAndDelete()` will immediately delete the comment, I wanted to first `findById()` and then check that the `comment.user === jwtAuth.user._id` before allowing it to `.remove()`.